alphafold3_pytorch/alphafold3.py

```diff
@@ -6012,8 +6012,12 @@ def forward(
                 is_nucleotide = is_rna | is_dna
                 is_polymer = is_protein | is_rna | is_dna
 
-                is_any_nucleotide_pair = to_pairwise_mask(is_nucleotide)
-                is_any_polymer_pair = to_pairwise_mask(is_polymer)
+                is_any_nucleotide_pair = einx.logical_and(
```

Comment:
ohh i see

could this and all the rest be rewritten as `repeat(is_type, '... j -> ... i j', i = is_type.shape[-1])` ?

Reply:
Maybe? It's hard for me to say without testing out a few inputs and outputs with this idea. The whole `torch.ones` thing I did was just to make absolutely sure that all atoms (represented as the rows in each of the constituent mask tensors) are considered when being paired up with e.g., a nucleotide atom or polymer atom, so as to not disqualify a "source atom" from being counted in `plddt_mask` just because it's a ligand or ion "source atom".

Reply:
yup, just a quick sanity check!

```python
import einx
from einops import repeat
import torch

is_something = torch.randint(0, 2, (2, 16)).bool()

one = einx.logical_and('... i, ... j -> ... i j', torch.ones_like(is_something), is_something)
two = repeat(is_something, '... j -> ... i j', i = is_something.shape[-1])

assert torch.allclose(one, two)
```

Reply:
Cool! Will push in a new PR

Reply:
hehe not a big deal

my bad for not understanding the intent!